src/Compilers/CSharp/Portable/Symbols/Metadata/PE/PEModuleSymbol.cs

```diff
@@ -691,7 +691,7 @@ internal override bool HasAssemblyRuntimeCompatibilityAttribute
         internal NamedTypeSymbol LookupTopLevelMetadataTypeWithNoPiaLocalTypeUnification(ref MetadataTypeName emittedName, out bool isNoPiaLocalType)
         {
             NamedTypeSymbol? result;
-            var scope = (PENamespaceSymbol?)this.GlobalNamespace.LookupNestedNamespace(emittedName.NamespaceSegments);
+            var scope = (PENamespaceSymbol?)this.GlobalNamespace.LookupNestedNamespace(emittedName.NamespaceSegmentsMemory);
```

Comment:
this is one of the heaviest hitters.  we used to break the FQN into N string pieces (each of which copies the data) for the namespace portion of the FQN.  Now we just have an array of ROM chunks.  The chunks are effectively free (lightweight structs), though we still pay for an array allocation here.  Ideally we could remove that as well with a specialized structure, but i kept that out of scope as i'm not seeing those arrays show up in traces with this PR.